code/MessageQueue.php

```diff
@@ -301,7 +305,11 @@ public static function consume_on_shutdown()
             switch (self::$onshutdown_option) {
                 case "sake":
                     $exec = Director::getAbsFile("framework/sake");
-                    `$exec MessageQueue_Process queue=$queue actions=$actions $limitClause $retriggerClause $stdout $stderr &`;
+		    if(Config::inst()->get('MessageQueue', 'foreground_process')){
+	                    `$exec MessageQueue_Process queue=$queue actions=$actions $limitClause $retriggerClause $stdout $stderr`;
+		    } else {
+			    `$exec MessageQueue_Process queue=$queue actions=$actions $limitClause $retriggerClause $stdout $stderr &`;
```

Comment:
This string `$exec MessageQueue_Process queue=$queue actions=$actions $limitClause $retriggerClause $stdout $stderr` is repeated 4 times.